Tests/SwiftStdlibTests/ArrayExtensionsTests.swift

```diff
@@ -36,6 +36,21 @@ final class ArrayExtensionsTests: XCTestCase {
         XCTAssertEqual(swappedEmptyArray, emptyArray)
     }
 
+    func testSortedLike() {
+        let expected1 = [1, 2, 3, 4, 5]
+        let candidate1 = [2, 5, 3, 1, 4]
+        XCTAssertEqual(candidate1.sorted(like: expected1, keyPath: \.self), expected1)
+
+        let candidate2 = [2, 5, 3, 6, 1, 4]
+        XCTAssertEqual(candidate2.sorted(like: expected1, keyPath: \.self), [1, 2, 3, 4, 5, 6])
+
+        // swiftlint:disable:next nesting
+        struct TestStruct { let prop: String }
```

Comment:
Also, what happens if the candidate size is smaller than the `like` array? Could you add a test case for that?